### ISSUE TITLE
MSVC compiler warnings

### ISSUE THREAD

**Issue by baconpaul** [free-audio/clap-wrapper#111]
in #81 we turned on `-Wall -Wextra -Wpedantic -Werror` for GCC and CLang

We have not done this for MSVC. We should do that for sure one day soon!

**Comment by defiantnerd** [comment]
Which compiler flags do we want exactly:

- /W4 Enable warning level 4 or even /Wall?
- /WX Treat warnings as errors

-Wextra and -Wpedantic have no equivalent in Visual Studio.

**Comment by baconpaul** [comment]
I have absolutely no idea! :)

**Comment by defiantnerd** [comment]
I have enabled them for a test and there are a few issues we need to solve first. Interesting enough, those issues should have been detected by GCC/Clang, too. ??

**Comment by baconpaul** [comment]
Yeah its nice to have multiple compilers so you can get the varied warnings!
